processor/src/main/resources/org/mapstruct/ap/internal/model/GeneratedType.ftl

```diff
@@ -14,6 +14,9 @@ package ${packageName};
 import ${importedType};
 </#list>
 
+<#if hasJavadoc()>
+<#nt><@includeModel object=javadoc/>
+</#if>
```

Comment:
I don't think that we need the `hasJavadoc` method. We can do it in a similar way as it is done for the constructor.

e.g.

```
<#if javadoc??><#nt> <@includeModel object=javadoc /></#if>
```